src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -66,12 +67,20 @@ bool GetEntrypointExecutableAbsolutePath(std::string& entrypointExecutable)
 
     // Get path to the executable for the current process using
     // platform specific means.
-#if defined(__LINUX__) || !defined(__APPLE__)
-    
-    // On non-Mac OS, return the symlink that will be resolved by GetAbsolutePath
-    // to fetch the entrypoint EXE absolute path, inclusive of filename.
-    entrypointExecutable.assign(symlinkEntrypointExecutable);
-    result = true;
+#if defined(__LINUX__)
+    // On Linux, fetch the entry point EXE absolute path, inclusive of filename.
+    char exe[PATH_MAX];
+    ssize_t res = readlink(symlinkEntrypointExecutable, exe, PATH_MAX - 1);
+    if (res != -1)
+    {
+        exe[PATH_MAX - 1] = (char) 0;
```

Comment:
Should the terminating zero be placed at [res] instead? The readlink doc says that it doesn't append the null.